source/common/grpc/common.cc

```diff
@@ -24,13 +24,63 @@ namespace Grpc {
 
 const std::string Common::GRPC_CONTENT_TYPE{"application/grpc"};
 
-void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& grpc_service,
-                        const std::string& grpc_method, bool success) {
+void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& protocol,
+                        const std::string& grpc_service, const std::string& grpc_method,
+                        const Http::HeaderEntry* grpc_status) {
+  if (!grpc_status) {
+    return;
+  }
+  uint64_t grpc_status_code;
+  bool success =
+      StringUtil::atoul(grpc_status->value().c_str(), grpc_status_code) && grpc_status_code == 0;
+  chargeStat(cluster, protocol, grpc_service, grpc_method, success);
+}
+
+void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& protocol,
+                        const std::string& grpc_service, const std::string& grpc_method,
+                        bool success) {
+  cluster.statsScope()
+      .counter(fmt::format("{}.{}.{}.{}", protocol, grpc_service, grpc_method,
+                           success ? "success" : "failure"))
```

Comment:
Why do we bin in the stats at this coarse granularity? It seems it would be generally useful to have a stats bucket for each gRPC status code, as it's useful when diagnosing issues to know if we're being rejected due to resource exhaustion, authentication, etc.